configurator/src/main/java/org/apereo/openequella/adminconsole/configurator/Configurator.java

```diff
@@ -0,0 +1,288 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apereo.openequella.adminconsole.configurator;
+
+import java.awt.Rectangle;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
+
+import javax.swing.BorderFactory;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.UIManager;
+
+import org.apereo.openequella.adminconsole.config.Config;
+import org.apereo.openequella.adminconsole.config.ServerProfile;
+import org.apereo.openequella.adminconsole.swing.ComponentHelper;
+import org.apereo.openequella.adminconsole.swing.ServerPicker;
+import org.apereo.openequella.adminconsole.swing.TableLayout;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Configurator extends JFrame implements WindowListener, ActionListener {
+
+    private static final long serialVersionUID = 4440091132569751172L;
+    private static final Logger LOGGER = LoggerFactory.getLogger(Configurator.class);
+
+    private static final int WINDOW_WIDTH = 450;
+
+    private final Config config;
+
+    private final ServerPicker serverPicker;
+    private final JButton addServerButton;
+    private final JButton editServerButton;
+    private final JButton removeServerButton;
+    private final JButton editProxyButton;
+
+    public static void main(String[] args) {
+        new Configurator();
+    }
+
+    public Configurator() {
+        try {
+            config = readConfig();
+            serverPicker = new ServerPicker(config);
+            addServerButton = new JButton("Add Server...");
+            editServerButton = new JButton("Edit Server...");
+            removeServerButton = new JButton("Remove Server");
+            editProxyButton = new JButton("Proxy Settings...");
+
+            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+
+            setTitle("Admin console configurator");
+            setResizable(false);
+            addWindowListener(this);
+            setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+            JPanel serverPanel = createServerPanel();
+            JPanel buttonsPanel = createButtonsPanel();
+
+            final int height1 = serverPanel.getMinimumSize().height;
+            final int[] rows = { height1, buttonsPanel.getPreferredSize().height };
+            final int[] cols = { WINDOW_WIDTH };
+
+            final TableLayout layout = new TableLayout(rows, cols);
```

Comment:
As this UI should not be very complicated, can we consider using other  layout such as GridLayout? I personally feel TableLayout may not be the best choice because of more configurations required.

Reply:
I think TableLayout is pretty simple to use.  I don't think it's worthwhile wrestling with GridLayout to make it look ok.